### PR TITLE
feat(trace-viewer): add network requests 'copy as cURL' button

### LINKED ISSUE #30082: [Feature]: add option export CURL from trace network
### 🚀 Feature Request

Added the option to export the CURL from playwright TRACE like in chrome devtools

### Example

When you want to resend a request there will be no need to manually reassemble it.

### Motivation

Option to resend / check failed requests.
The option exists in most browsers.

image

### PR DESCRIPTION
**Pull request by mxschmitt**
Fixes https://github.com/microsoft/playwright/issues/30082

Preview: 

(looks like a bug in macOS that no cursor: pointer is getting shown - there is one when I don't record.)

https://github.com/user-attachments/assets/75db3c0a-cf31-46e6-a329-bd2836003307

### CODE REVIEW

File: packages/trace-viewer/src/ui/networkTab.tsx [hunk 1]
```diff
@@ -371,3 +379,120 @@ function filterEntry({ searchValue, resourceType }: FilterState) {
     return typePredicate(entry.contentType) && entry.name.url.toLowerCase().includes(searchValue.toLowerCase());
   };
 }
+
+// The following function is derived from Chromium's source code
```
Comment (dgozman) [inline packages/trace-viewer/src/ui/networkTab.tsx:383]: Put this into a separate `third_party/curl.ts` file with a proper copyright instead.

File: packages/trace-viewer/src/ui/networkTab.tsx [hunk 2]
```diff
@@ -371,3 +379,120 @@ function filterEntry({ searchValue, resourceType }: FilterState) {
...
+
+  // cURL command expected to run on the same platform that DevTools run
+  // (it may be different from the inspected page platform).
+  const escapeString = platform === 'win' ? escapeStringWin : escapeStringPosix;
+
+  command.push(escapeString(request.request.url).replace(/[[{}\]]/g, '\\$&'));
+
+  let inferredMethod = 'GET';
+  const data = [];
+  // TODO: read actual request/response payload
```
Comment (dgozman) [inline packages/trace-viewer/src/ui/networkTab.tsx:465]: What does this TODO mean?

File: packages/web/src/components/listView.tsx [hunk 3]
```diff
@@ -170,6 +208,29 @@ export function ListView<T>({
           {typeof rendered === 'string' ? <div style={{ textOverflow: 'ellipsis', overflow: 'hidden' }}>{rendered}</div> : rendered}
         </div>;
       })}
+      {contextMenu && (
```
Comment (dgozman) [inline packages/web/src/components/listView.tsx:211]: Showing a proper context menu is a complicated business. Perhaps we should use [popover](https://developer.mozilla.org/en-US/docs/Web/API/Popover_API/Using)?

File: packages/web/src/components/toolbarButton.tsx [hunk 4]
```diff
@@ -40,11 +41,8 @@ export const ToolbarButton: React.FC<React.PropsWithChildren<ToolbarButtonProps>
   testId,
   className,
 }) => {
-  className = (className || '') + ` toolbar-button ${icon}`;
-  if (toggled)
-    className += ' toggled';
   return <button
-    className={className}
+    className={clsx(className, `toolbar-button ${icon}`, toggled && 'toggled')}
```
Comment (mxschmitt) [inline packages/web/src/components/toolbarButton.tsx:45]: drive-by cleanup

File: packages/trace-viewer/src/ui/networkResourceDetails.tsx [hunk 5]
```diff
@@ -90,6 +92,13 @@ const RequestTab: React.FunctionComponent<{
     </> : null}
     <div className='network-request-details-header'>Request Headers</div>
     <div className='network-request-details-headers'>{resource.request.headers.map(pair => `${pair.name}: ${pair.value}`).join('\n')}</div>
+    <div className='network-request-details-header'>Copy Request</div>
+    <div className='network-request-details-copy'>
+      As cURL: <CopyToClipboard description='Copy as cURL' value={() => generateCurlCommand(resource)}/>
+    </div>
+    <div className='network-request-details-copy'>
+      As Fetch (Browser): <CopyToClipboard description='Copy as Fetch' value={() => generateFetchCall(resource)}/>
```
Comment (dgozman) [inline packages/trace-viewer/src/ui/networkResourceDetails.tsx:100]:
```suggestion
      As Fetch: <CopyToClipboard description='Copy as Fetch' value={() => generateFetchCall(resource)}/>
```

File: packages/web/src/components/toolbarButton.tsx (hunk 4 above)
Comment (dgozman) [inline packages/web/src/components/toolbarButton.tsx:45]:
```suggestion
    className={clsx(className, 'toolbar-button', icon, toggled && 'toggled')}
```

**Review by dgozman — APPROVED** [review]
